feat (replays/33273): added project column to the replays screen

Danondso opened the pull request:
# Description
#33273 
- Added the project column to the replays screen

# Screenshots
<img width="286" alt="image" src="https://user-images.githubusercontent.com/7014871/163438457-6715f10a-2906-4c1a-8ae6-36e160a111e8.png">
<img width="1663" alt="image" src="https://user-images.githubusercontent.com/7014871/163438490-3b3997c0-414c-4f6d-9f35-b29f7a19bed6.png">

# Known Gotchas
- none

billyvg committed:
feat(ci): Change Sentry release when running pytest (#33597)

When running pytest, we use [pytest-sentry](https://github.com/getsentry/pytest-sentry/) to track flakey tests (among others). By default the python SDK will try to find a default release, in this case, was the commit SHA. This release version was conflicting with the version we use for CI/CD, and also associating it with our javascript+sentry projects for those prod releases. This is confusing because it looks like we are making empty releases.

We should instead define a different version string when we use Sentry in our CI env.

Danondso committed:
proto: project column with name and link, still needs correct icon

Zylphrex committed:
fix(profiling): Set flamegraph origin to beneath timeline (#33599)

The inverted flamegraph has the origin set to (0,0) and this resulted in various
off by 1 errors that required adjustment. This change ensures that the origin
for the inverted flamegraph also accounted for the timeline. This means the
various adjustments could be removed from.

k-fish committed:
fix(perf): Fix error messages coming from Gen. Query (#33341)

The error messages don't always return in the same format, so this is actually an older bug. This will check whether the error is being returned as a string or an object {code, message, extra} with room to expand this in the future if we find more cases.

armenzg committed:
fix(ci): Add visual-diff with workflow_run for forks (#33609)

`visual-diff.yml` got removed in #33455
It allowed running Visual Snapshots on forks.

matejminar committed:
chore(dashboards): Add dashboards-mep flag (#33611)

vuluongj20 committed:
fix(hovercard): Fix arrow position with non-default offset (#33601)

Danondso committed:
feat: added the project column with the correct badges for projects

Danondso committed:
fix: removed unused platform field from query and types.

narsaynorath committed:
feat(top-events): Support excluding other from topEvents query (#33606)

ryan953 commented on static/app/views/replays/replays.tsx:
```diff
@@ -61,7 +63,7 @@ class Replays extends React.Component<Props> {
       <React.Fragment key={replay.id}>
         <Link
           to={`/organizations/${organization.slug}/replays/${generateEventSlug({
-            project: replay['project.name'],
```
good change, looks like `project.name` wasn't part of `static/app/views/replays/types.tsx` before either.

vuluongj20 committed:
ref(page-filters): Move issues sort options inside table (#33590)

narsaynorath committed:
fix(new-widget-builder-experience): Unnecessary dashboard visit requests (#33619)

Danondso commented on static/app/views/replays/replays.tsx:
```diff
@@ -79,6 +81,18 @@ class Replays extends React.Component<Props> {
             displayEmail={replay.url?.split('?')[0] || ''}
           />
         </Link>
+        <Projects orgId={organization.slug} slugs={[replay.project]}>
```
I'll find out here and adjust accordingly.

Danondso replied:
Doesn't seem to be the case as far as I can tell. <Projects /> leverages a reflux store that stores projects that have already been fetched along with managing calls to the API for getting any projects that don't yet exist in the store.

That call to `/projects` there is the only one that occurs for loading up the replays screens. A refresh will cause it to pull again but visiting other pages and coming back doesn't trigger the call.
<img width="433" alt="image" src="https://user-images.githubusercontent.com/7014871/163455136-b0bfae52-4a4f-40a3-b6be-0bd77a590b82.png">

k-fish committed:
ref(perf): Move long-tasks to app-wide (#33614)

Part 2 of a previous PR to show long tasks vs. interactions, this moves long-tasks to be app-wide now that they've been tested on prod on a slice of the app (event view). Long tasks should cover catching any rendering or frontend performance issues generically, since anything that is a long task means that the user is likely seeing jank or worse, is CPU bound.

Commit:
ref(page-filters): All environments -> All env (#33568)

* all env change

* fix test

mikellykels committed:
feat(workflow): Add Undo to issue removal toast (#33249)

* feat(workflow): Add Undo to issue removal toast

Add Undo to issue removal toast.

FIXES WOR-1728

* add unto to mark reviewed

* fix mark reviewed undo

* update status and cancel inflight requests on undo

* remove undo for mark reviewed and set status as unresolved for ignored and resolved when clicking undo

* move Group type

Commit:
feat(onboarding): remove welcome page experiment and use new experience (#33616)

This PR copies the welcome page component from the targeted onboarding flow into the default onboarding flow and removes the TargetedOnboardingWelcomePageExperimentV2 experiment. There are some minor differences to handle the different prop types but everything else is the same.

vuluongj20 committed:
ref(alerts): Use Badge in team filter (#33596)

shruthilayaj committed:
feat(dashboards): Update Release Health dataset to use sessions v2 (#33582)

Switching out Release Health to use sessions v2.
This does not completely replace all the functionality
available on the metrics endpoint (including filtering
and order by).

ryan953 commented on static/app/views/replays/types.tsx:
```diff
@@ -1,6 +1,7 @@
 export type Replay = {
```
More TS shenanigans.. .
`projectID` isn't fetched, and `replayId` seems to be empty / not used any of the time.

Danondso replied:
Oh good catch I'll get those removed.

Danondso replied:
Alright removed those.

k-fish committed:
feat(perf): Add MEP to transaction summary (#32773)

* feat(perf): Add MEP to transaction summary

This adds the query param to the duration and sidebar api requests as well as some preliminary UI to switch sampled/unsampled states in the transaction summary.

maxiuyuan committed:
fix(sudo modal): no longer trying to log superuser access on sudo (#33622)

mikellykels committed:
feat(workflow): Tidy up projects page design (#33608)

* feat(workflow): Tidy up projects page design

Tidy up projects page design and update for mobile views. Also properly sort projects.

FIXES WOR-1784

* update test

* fix styles

* update padding

* remove min-width

ryan953 committed:
feat(replay): Implement the View in Fullscreen button (#33566)

Implement the 'View in full screen' button that shows the player, as well as the play/pause controls.

Danondso committed:
fix: removed a couple more unused properties from the Replay type and fields array

Danondso commented:
I goofed on my merging and wrecked the diff, addressing that before opening this again.
